gamemodes/terrortown/gamemode/client/cl_targetid.lua

```diff
@@ -172,13 +172,18 @@ function GM:PostDrawTranslucentRenderables(bDrawingDepth, bDrawingSkybox)
 		local ply = plys[i]
 		local rd = ply:GetSubRoleData()
 
-		if ply:IsActive()
+		local shouldDrawDefault = ply:IsActive()
 			and ply:HasRole()
 			and (not client:IsActive() or ply:IsInTeam(client) or rd.isPublicRole)
 			and not rd.avoidTeamIcons
-		then
-			DrawOverheadRoleIcon(ply, rd.iconMaterial, ply:GetRoleColor())
-		end
+
+		local shouldDraw, material, color = hook.Run("TTT2ModifyOverheadIcon", ply, shouldDrawDefault)
+
+		if shouldDraw == false
+			or not shouldDrawDefault and not material and not color
+		then continue end
```

Comment:
Shouldnt this be combined when one or the other is missing?
Not really sure if this works
```suggestion
		if shouldDraw == false
			or not shouldDrawDefault and not (material and color)
		then continue end
```

Reply:
your suggestion should use an `or`

Reply:
I designed it in a way that both have to be set, otherwise it could end up in an OHI without icon or the other way around

Reply:
Okay, then assume this case:
shouldDrawDefault = false
material = MaterialX
color = nil
then it doesnt continue, because:
or not shouldDrawDefault = true
and not material = false
and (not color = true) = false , so it stays false

Reply:
I think this is okay
if you only return one argument even though you are supposed to return two, this is fine

what should happen instead?

Reply:
oh okay in this case you'd be okay. Id thought color and material need to be given 😄

Reply:
DId you mistype? Yes, both have to be returned, just as I said

Reply:
Well in the current state it is continued aka not drawn, when only both material and icon are missing. When only a material is given or only a color, then it is not continued and it will be drawn.
If that should be like that, then I resolve this conversation 😄

Reply:
you're absolutely right!